packages/core/src/anchor/anchor-processing-loop.ts

```diff
@@ -111,4 +124,18 @@ export class AnchorProcessingLoop {
     this.#anchorPollingMetrics.stopPublishingStats()
     return this.#loop.stop()
   }
+
+  /**
+   * Check the poll time for a stream. Updates the poll time and returns true if the stream has not been polled in the
+   * last pacing interval.
+   */
+  checkPollTime(streamId: string): boolean {
+    if (this.#pollCache.has(streamId) && (Date.now() < this.#pollCache.get(streamId))) {
+      return false
+    }
+    // Add some jitter to prevent all streams from being polled at exactly the same time after a restart
```

Comment:
this seems like it only helps after the first pass over the AnchorRequestStore.  The first time through though you're going to poll everything in the store as fast as possible because nothing will be in the cache.

Reply:
Yes, agreed. Only helps after the first time through the store. Adding jitter is a good idea anyway because often traffic patterns will start lining up and causing bursty processing.

Reply:
That's actually why I wanted to store the last polled timestamp in the DB versus keeping it in memory.

There's always the risk that at startup we slam the CAS too hard. Realistically, we might ask partners to upgrade and some of the upgrades across the network happen around the same time. Now, a bunch of Ceramic nodes will start slamming CAS with everything in their anchor stores.

Probably not a very likely scenario so if the risks of using the DB outweigh the risks of something like this happening, I'm cool with keeping the cache in-memory.

Reply:
Although, if the node takes long enough to restart, it won't matter even if stuff is written to disk... All the timestamps would have expired anyway 😭

Reply:
Also, it's unlikely that partners would all restart nodes at the SAME time. Within a few hours, sure. A few minutes in the extreme. But nearing the same second? Hard to believe.

Reply:
Yes, that's fair.

We should still figure out a nicer way to pace status requests at startup since we could slam the CAS with 100s of TPS from a single node. This would be good to avoid even if CAS can take it.

Reply:
yeah, this leads to still thinking our best bet is a small sleep between each poll in just the first pass through the store after restart.